Source/Android/jni/Host.h

```diff
@@ -12,13 +12,26 @@
 // sequentially for access.
 struct HostThreadLock
 {
-  static std::mutex s_host_identity_mutex;
-  std::unique_lock<std::mutex> m_lock;
-
+public:
   explicit HostThreadLock() : m_lock(s_host_identity_mutex) { Core::DeclareAsHostThread(); }
   HostThreadLock(const HostThreadLock& other) = delete;
   HostThreadLock(HostThreadLock&& other) = delete;
   HostThreadLock& operator=(const HostThreadLock& other) = delete;
   HostThreadLock& operator=(HostThreadLock&& other) = delete;
-  ~HostThreadLock() { Core::UndeclareAsHostThread(); }
+
+  void Lock()
+  {
+    m_lock.lock();
+    Core::DeclareAsHostThread();
+  }
+
+  void Unlock()
+  {
+    m_lock.unlock();
+    Core::UndeclareAsHostThread();
+  }
```

Comment:
Asserts in here wouldn't hurt, probably. So an `ASSERT(!IsHostThread())` in lock and a `ASSERT(IsHostThread())` in unlock.

Reply:
You also removed the destructor here, that seems wrong...?

Reply:
You'll get an exception from the lock if you call `lock` while already locked or vice versa, so I don't think it's necessary.

I meant to just move the destructor, but I must have deleted it by mistake in a rebase conflict. I'll fix that.

Reply:
You'll get an exception from the lock if you call `lock` while already locked or vice versa, so I don't think it's necessary.

I meant to just move the destructor, but I must have deleted it by mistake in a rebase conflict. I'll fix that.

Reply:
Ah, okay yeah if it errors out anyway then that's fine.